distribution/src/resources/yarn-drillbit.sh

```diff
@@ -110,6 +114,36 @@
 #     Enables Java GC logging. Passed from the drill.yarn.drillbit.log-gc
 #     garbage collection option.
 
+### Function to enforce CGroup (Refer local drillbit.sh)
+check_and_enforce_cgroup(){
+    dbitPid=$1;
+    kill -0 $dbitPid
+    if [ $? -gt 0 ]; then 
+      echo "ERROR: Failed to add Drillbit to CGroup ( $DRILLBIT_CGROUP ) for 'cpu'. Ensure that the Drillbit ( pid=$dbitPid ) started up." >&2
+      exit 1
+    fi
+    SYS_CGROUP_DIR=${SYS_CGROUP_DIR:-"/sys/fs/cgroup"}
+    if [ -f $SYS_CGROUP_DIR/cpu/$DRILLBIT_CGROUP/cgroup.procs ]; then
+      echo $dbitPid > $SYS_CGROUP_DIR/cpu/$DRILLBIT_CGROUP/cgroup.procs
+      # Verify Enforcement
+      cgroupStatus=`grep -w $pid $SYS_CGROUP_DIR/cpu/${DRILLBIT_CGROUP}/cgroup.procs`
+      if [ -z "$cgroupStatus" ]; then
```

Comment:
You're right. I seem to have missed negating the check. Since this check only affects publication of a message and not the actual application of the CGroup, we didn't catch it during testing. I'll fix this port and the original patch as well.

Reply:
Fixed the changes.